cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java

```diff
@@ -217,6 +224,35 @@ public void finishExecutable(@Param(ScoreLangConstants.RUN_ENV) RunEnvironment r
         }
     }
 
+    public void canExecute(@Param(ScoreLangConstants.RUN_ENV) RunEnvironment runEnv,
+                           @Param(EXECUTION_RUNTIME_SERVICES) ExecutionRuntimeServices executionRuntimeServices,
+                           @Param(ScoreLangConstants.NODE_NAME_KEY) String nodeName,
+                           @Param(ScoreLangConstants.NEXT_STEP_ID_KEY) Long nextStepId) {
+        try {
+            if (!StringUtils.isEmpty(runEnv.getExecutionPath().getParentPath())) {
+                //If it is start of a sub flow then the check should not happen
+                runEnv.putNextStepPosition(nextStepId);
+                return;
+            }
+
+            while (!executionPreconditionService.canExecute(valueOf(executionRuntimeServices.getExecutionId()))) {
+                try {
+                    logger.warn("Execution precondition not fulfilled. Waiting for it to be true.");
```

Comment:
I don't think anyone would understand this log. Log the execution id and I would let the executionPreconditionService to log this so we can get more details on the exact reason why the execution can not proceed.

Reply:
If we only want the ID to be logged then we could just remove this log entirely, since the ID is logged afterwards by the service anyway